pkg/didmethod/peer/resolver.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package peer
+
+import (
+	errors "golang.org/x/xerrors"
+)
+
+// DIDResolver resolver
+type DIDResolver struct {
+	store *DIDStore
+}
+
+// NewDIDResolver new Peer DID resolver instance
+func NewDIDResolver(store *DIDStore) *DIDResolver {
+	return &DIDResolver{store: store}
+}
+
+// Read implements DidMethod.Read function
+func (resl *DIDResolver) Read(did string, versionID interface{}, versionTime string, noCache bool) ([]byte, error) {
```

Comment:
There are extra unused arguments. I assume there should be a TODO describing how they will be used.

Also are they intended to be mandatory or optional?

Reply:
These are all [optional](https://w3c-ccg.github.io/did-resolution/#resolving-input) (except for the DID of course)

Reply:
Opened #65 to refactor this